components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -258,8 +263,12 @@ public void updatePassword(String code, String password, Property[] properties)
 
     }
 
-    private void checkPasswordValidity(UserStoreException e) throws IdentityRecoveryClientException {
+    private void checkPasswordValidity(UserStoreException e, int tenantId) throws IdentityRecoveryClientException, IdentityRecoveryServerException {
```

Comment:
Can we throw IdentityRecoveryException rather throwing IdentityRecoveryClientException and IdentityRecoveryServerException ?

Reply:
throwing IdentityRecoveryClientException was the default behaviour. shall we keep it as it is?